Gems/AWSCore/Code/Tools/ResourceMappingTool/utils/environment_utils.py

```diff
@@ -38,6 +39,19 @@ def setup_qt_environment(bin_path: str) -> None:
     new_path = os.pathsep.join([binaries_path, path])
     os.environ['PATH'] = new_path
 
+    # On Linux, we need to load pyside2 and related modules as well
+    if platform.system() == 'Linux':
+        import ctypes
+
+    preload_shared_libs = [f'{bin_path}/libpyside2.abi3.so.5.14',
+                           f'{bin_path}/libQt5Widgets.so.5']
+
+    for preload_shared_lib in preload_shared_libs:
+        if not os.path.exists(preload_shared_lib):
+            logger.error(f"Cannot find required shared library at {preload_shared_lib}")
+            exit(-1)
```

Comment:
will application get exited properly? I think this just exit the function?

Reply:
no, exit the program. *return* exits the function only

Reply:
Minor: prefer return here?

Reply:
We need to halt because the program will crash when attempting to import the pyside packages. exit(-1) was used here because it is what the calling tool script does:

```
    try:
        logger.info("Importing tool required modules ...")
        from PySide2.QtCore import Qt
        from PySide2.QtWidgets import QApplication
        from manager.configuration_manager import ConfigurationManager
        from manager.controller_manager import ControllerManager
        from manager.thread_manager import ThreadManager
        from manager.view_manager import ViewManager
        from style import azqtcomponents_resources
        from style import editormainwindow_resources
    except ImportError as e:
        logger.error(f"Failed to import module [{e.name}] {e}")
        environment_utils.cleanup_qt_environment()
        exit(-1)
```

I could remove the exit(-1) and then just let the exception happen when the import is attempted